src/Workspaces/Core/Portable/Workspace/ProjectSystem/FileWatchedReferenceFactory.cs

```diff
@@ -198,3 +167,52 @@ private void FileReferenceChangeContext_FileChanged(object? sender, string fullF
         }
     }
 }
+
+internal static class FileWatchedReferenceFactory
+{
+    public static FileWatchedReferenceFactory<AnalyzerFileReference> CreateAnalyzerReferenceFactory(IFileChangeWatcher fileChangeWatcher)
+        => new(fileChangeWatcher, static () => []);
+
+    public static FileWatchedReferenceFactory<PortableExecutableReference> CreateMetadataReferenceFactory(IFileChangeWatcher fileChangeWatcher)
+        => new(fileChangeWatcher, getAdditionalWatchedDirectories: static () =>
+        {
+
+            using var _ = PooledHashSet<string>.GetInstance(out var referenceDirectories);
+
+            // On each platform, there is a place that reference assemblies for the framework are installed. These are rarely going to be changed
```

Comment:
made this logic specific to the PortableExecutableReference, but i don't actually know if that's what we want.